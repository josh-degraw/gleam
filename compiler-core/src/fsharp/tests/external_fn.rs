use crate::assert_fsharp;

#[test]
fn external_fn_from_std_lib() {
    assert_fsharp!(
        r#"
@external(fsharp, "FSharp.MyApp", "run")
pub fn run() -> Int

pub fn foo() -> Int {
    run()
}
"#
    );
}

#[test]
fn external_fn_from_file() {
    assert_fsharp!(
        r#"
@external(fsharp, "./SomeFile.fs", "SomeFile.run")
pub fn run() -> Int

pub fn foo() -> Int {
    run()
}
"#
    );
}
