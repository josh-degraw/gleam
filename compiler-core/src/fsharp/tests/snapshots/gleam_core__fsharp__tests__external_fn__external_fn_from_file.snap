---
source: compiler-core/src/fsharp/tests/external_fn.rs
expression: "\n@external(fsharp, \"./SomeFile.fs\", \"SomeFile.run\")\npub fn run() -> Int\n\npub fn foo() -> Int {\n    run()\n}\n"
---
module rec my.``mod``
let run (): int = SomeFile.run ()
let foo (): int = begin
    run()
end
